src/services/ghost/ContextAnalyzer.ts

```diff
@@ -0,0 +1,227 @@
+import * as vscode from "vscode"
+import { GhostSuggestionContext } from "./types"
+import { ContextAnalysis, UseCaseType } from "./types/PromptStrategy"
+
+/**
+ * Analyzes GhostSuggestionContext to determine the appropriate use case and context properties
+ */
+export class ContextAnalyzer {
+	/**
+	 * Analyzes the given context to determine use case and properties
+	 * @param context The suggestion context to analyze
+	 * @returns Analysis result with use case and context properties
+	 */
+	analyze(context: GhostSuggestionContext): ContextAnalysis {
```

Comment:
Hmm I wonder if this class is doing too much? I guess multiple strategies could care about the same `analysis` criteria so it makes sense to put all the logic here instead of putting inside the strategies themselves?